flow/project.py

```diff
@@ -1870,52 +1969,105 @@ def _print_progress(x):
+                        file=err))
                 else:
                     raise RuntimeError("Configuration value status_parallelization is invalid. "
-                                       "You can set it to 'thread', 'parallel', or 'none'"
-                                       )
+                                       "You can set it to 'thread', 'parallel', or 'none'")
             except RuntimeError as error:
                 if "can't start new thread" not in error.args:
                     raise   # unrelated error
 
-                t = time.time()
-                num_jobs = len(jobs)
-                statuses = []
-                for i, job in enumerate(jobs):
-                    statuses.append(_get_job_status(job))
-                    if time.time() - t > 0.2:  # status interval
-                        print(
-                            'Collecting job status info: {}/{}'.format(i+1, num_jobs),
-                            end='\r', file=err)
-                        t = time.time()
-                # Always print the completed progressbar.
-                print('Collecting job status info: {}/{}'.format(i+1, num_jobs), file=err)
-                return statuses
-
-    def _fetch_status_in_parallel(self, pool, pickle, jobs, ignore_errors, cached_status):
+                def print_status(iterable, fetch_status, description):
+                    t = time.time()
+                    num_itr = len(iterable)
+                    results = []
+                    for i, itr in enumerate(iterable):
+                        results.append(fetch_status(itr))
+                        # The status interval 0.2 seconds is used since we expect the
+                        # status for an aggregate to be fetched within that interval
+                        if time.time() - t > 0.2:
+                            print(f'{description}: {i+1}/{num_itr}', end='\r', file=err)
+                            t = time.time()
+                    # Always print the completed progressbar.
+                    print(f'{description}: {i+1}/{num_itr}', file=err)
+                    return results
+
+                label_results = print_status(distinct_jobs, get_job_labels,
+                                             "Collecting job label info")
+                op_results = print_status(singleton_groups, get_group_status,
+                                          "Collecting operation status")
+
+        results, index, i = list(), dict(), 0
+        for job in distinct_jobs:
+            results_entry = dict()
+            results_entry['job_id'] = str(job)
+            results_entry['operations'] = dict()
+            results_entry['_operations_error'] = None
+            results_entry['labels'] = list()
+            results_entry['_labels_error'] = None
+            results.append(results_entry)
+            index[job.get_id()] = i
+            i += 1
```

Comment:
We require the results to be a list of dictionaries because of the logic we follow in `print_status`.

Reply:
That seems like perhaps a more extensive change since it will require modifying our status renderer class etc. Can you make a new issue to track that we want to change this, and then follow up on that in another PR once this is done?